### PR TITLE
Fix the error thrown on enabling the "OpenId.Google" feature

### PR DESCRIPTION
**Pull request by ThaerAlAjlouni**
Remove the **tilde ~** from the `LogonCallbackUrl` relative path to prevent errors after enabling the `OpenId.Google` feature.

This will fix issue #7546

### CODE REVIEW

**Comment by infofromca** [comment]
I think this just resovled the half.
because the user can still input ~/Users/Account/LogonCallback to the LogonCallbackUrl textbox.
My idea is to disable user input in the textbox of LogonCallbackUrl

**Comment by ThaerAlAjlouni** [comment]
I think adding validation will be better than not allowing the user to edit the callback URL.

File: src/Orchard.Web/Modules/Orchard.OpenId/Constants/General.cs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿namespace Orchard.OpenId.Constants {
     public class General {
         public const string AuthenticationErrorUrl = "~/Authentication/Error";
-        public const string LogonCallbackUrl = "~/Users/Account/LogonCallback";
+        public const string LogonCallbackUrl = "/Users/Account/LogonCallback";
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.OpenId/Constants/General.cs]: Have you tried on multi-tenancy with a route prefix in this case?

**Comment by sebastienros** [comment]
@infofromca is the PR ok with you now?

File: src/Orchard.Web/Modules/Orchard.OpenId/Constants/General.cs (hunk 1 above)
Comment (ThaerAlAjlouni) [inline src/Orchard.Web/Modules/Orchard.OpenId/Constants/General.cs]: No to be honest, I haven't test it on multi-tenancy. I will test it and update the PR if required.

Comment (ThaerAlAjlouni) [inline src/Orchard.Web/Modules/Orchard.OpenId/Constants/General.cs]: @sebastienros, PR updated after modifying code to support multi-tenancy

**Comment by infofromca** [comment]
I guess the purpose of GetCallbackPath can be found somewhere in the old code.
but it is difficult to find.
long time ago, I suggested that we moved common use code to  a central module ( or Utility project or folder) , if it is not easy for orchard 1 , we should consider it in orchard 2

**Comment by sebastienros** [comment]
I think the `IsValid` property should be removed from the class, or at least converted to a method so that is doesn't get serialized in the db.

**Comment by sebastienros** [comment]
@infofromca this is handled automatically in ASP.NET Core (hence in Orchard Core) because they introduced a `PathBase` in the request object. And everything is local to this. So we assign the tenant prefix now.

File: src/Orchard.Web/Modules/Orchard.OpenId/OwinMiddlewares/Google.cs [hunk 2]
```diff
@@ -27,7 +29,7 @@ public class Google : IOwinMiddlewareProvider {
             var authenticationOptions = new GoogleOAuth2AuthenticationOptions {
                 ClientId = settings.ClientId,
                 ClientSecret = settings.ClientSecret,
-                CallbackPath = new PathString(settings.CallbackPath)
+                CallbackPath = new PathString(GetCallbackPath(workContext, settings))
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.OpenId/OwinMiddlewares/Google.cs]: Isn't it fine if the users adds the prefix by themself? We can just document that the url they enter needs to contain the tenant prefix if there is one. I would prefer this option if it works.

**Comment by sebastienros** [comment]
@infofromca what do you think about my comments, does it solve your concerns?

**Comment by infofromca** [comment]
the code is ok 
because we do not have PathBase right now.
ask user to input tenant prefix to too technical, and not every editor is developer.
IsValid seems not used in db

File: src/Orchard.Web/Modules/Orchard.OpenId/OwinMiddlewares/Google.cs (hunk 2 above)
Comment (ThaerAlAjlouni) [inline src/Orchard.Web/Modules/Orchard.OpenId/OwinMiddlewares/Google.cs]: Why to add an extra step of responsibilty on the user?
I prefer less intervention from the user in such scenarios.

File: src/Orchard.Web/Modules/Orchard.OpenId/Views/EditorTemplates/Parts.AzureActiveDirectorySettings.cshtml [hunk 3]
```diff
@@ -2,31 +2,31 @@
 
 <h2>Azure Active Directory Settings</h2>
 <fieldset>
-    @Html.LabelFor(m => m.Tenant, T("Tenant"))
+    <label for="AzureActiveDirectorySettings_Tenant">@T("Tenant")</label>
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.OpenId/Views/EditorTemplates/Parts.AzureActiveDirectorySettings.cshtml]: `@Html.IdFor()` everywhere please

Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.OpenId/Views/EditorTemplates/Parts.AzureActiveDirectorySettings.cshtml]: :up::date: ?

Comment (ThaerAlAjlouni) [inline src/Orchard.Web/Modules/Orchard.OpenId/Views/EditorTemplates/Parts.AzureActiveDirectorySettings.cshtml]: I will try to do it the upcoming Friday.

**Comment by sebastienros** [comment]
As a reminder, anyone who has write access to the repository is allowed to help some PRs when they are blocked by some small patches. I'll try to remind it during tomorrow's meeting.

File: src/Orchard.Web/Modules/Orchard.OpenId/Views/EditorTemplates/Parts.AzureActiveDirectorySettings.cshtml (hunk 3 above)
Comment (ThaerAlAjlouni) [inline src/Orchard.Web/Modules/Orchard.OpenId/Views/EditorTemplates/Parts.AzureActiveDirectorySettings.cshtml]: @sebastienros Done

**Pull request merged**